consul/state/catalog_test.go

```diff
@@ -10,14 +10,23 @@ import (
 	"github.com/hashicorp/consul/lib"
 	"github.com/hashicorp/consul/types"
 	"github.com/hashicorp/go-memdb"
+	uuid "github.com/hashicorp/go-uuid"
 )
 
+func makeRandomNodeID(t *testing.T) types.NodeID {
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	return types.NodeID(id)
+}
+
 func TestStateStore_EnsureRegistration(t *testing.T) {
```

Comment:
I think it's worth it to add another test, or a clause here that calls `NodeServices()` with a UUID.